upt_macports/upt_macports.py

```diff
@@ -105,7 +106,23 @@ def _pkgname(self):
 
     @staticmethod
     def _normalized_macports_name(name):
-        return name
+        return name.replace('::', '-')
+
+    def _cpandir(self):
+        pkg = self.upt_pkg
+        if pkg.archives:
+            archive_name = pkg.archives[0].url.split('/')[-1]
+            part_name = pkg.name.replace('::', '-').split('-')[0]
+            check_url = f'https://cpan.metacpan.org/modules/by-module/{part_name}/{archive_name}' # noqa
+            r = requests.head(check_url)
+            if r.status_code == 200:
+                print(r)
+                return ''
+            else:
+                fallback_dist = '/'.join(pkg.archives[0].url.split('id/')[1].split('/')[:-1]) # noqa
+                if fallback_dist:
+                    return f' ../../authors/id/{fallback_dist}/'
+        return ' # could not locate dist file'
```

Comment:
So, this function deserves a big wall of text (or multiple ones) to explain what is happening. Also, I would rather eliminate the weirdest case first, like people do in C. In C, you often see:

```c
void foo()
{
    if (something_weird())
        return;

    if (something_weird_too())
        return;

    /* And now, for the real stuff */
    ...
}
```

You get into the "main" part of the function with less context to remember, and you gain a level of indentation. So, here, we could write:

```python
    if not pkg.archives:
        return ' # No dist file'

    # We start by checking the usual URL
    blabla
    if r:
        return ''
    else:
        # Sometimes, it is not available, in which case we fall back to the following:
        return f'...'
```

What do you think?

Reply:
Yeah agree this makes more sense, will push the changes in some time.